src/dist/shifted.rs

```diff
@@ -0,0 +1,242 @@
+use crate::data::ShiftedSuffStat;
+use crate::traits::*;
+use rand::Rng;
+#[cfg(feature = "serde1")]
+use serde::{Deserialize, Serialize};
+
+/// A wrapper for distributions that adds a dx parameter
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "serde1", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde1", serde(rename_all = "snake_case"))]
+pub struct Shifted<D> {
+    parent: D,
+    dx: f64,
```

Comment:
Is it worth making this generic so we can implement this for distributions with other types of support?

Reply:
Ideally yes. But I had a go and it was just a big mess. Let's talk through it some time if you have ideas for the right way to approach this

Reply:
Sounds good.